### ISSUE TITLE
Feature: Enable Dark/Light theme based on OS theme

### ISSUE THREAD

**Issue by jjonathan313** [a85/POSTMan-Chrome-Extension#5817]
It would be nice to match the theme of Postman to the theme enabled on the host OS (macOS and Windows 10), maybe as a third option in themes.

**Comment by a85** [comment]
There are no plans to implement this but we will build more customizability for themes soon. Until then users should set the themes themselves.

**Comment by augsteyer** [comment]
is it possible to consider this to be placed on the feature board with at least lowest priority? It's very inconvenient to switch this twice a day. If it's a tricky task to do, maybe consider adding a shortcut for this theme switch toggle?

**Comment by tyleroconnell** [comment]
@a85 is there anything the community can do to help make this feature a reality?

**Comment by falnyr** [comment]
@a85 bump

**Comment by dveldhoen** [comment]
+1 for this

**Comment by mistryvipul2002** [comment]
+1 for this feature

**Comment by rparias** [comment]
It would be a really good feature! Thinking on the user (developer) experience.

**Comment by downdrown** [comment]
+1 ... my eyes demand this feature!

**Comment by proshin-roman** [comment]
+1 That's just a nightmare: I have a small app sitting in the tray that I can just click, and the whole system switches between light and dark modes. I usually switch to the dark mode in the evening, and all apps (I'm a web java developer) switch to the dark theme too except one - Postman. This has already prompted me to explore other apps similar to Postman. And I want to say: Postman is the best so far - the number of features and how it works all together. But there are other options, e.g. I'm trying to switch to the embedded HTTP client provided by IntelliJ IDEA. That might be quite some work since I have many collections stored in Postman, but my eyes deserve it, I hope :) So I'm really looking forward to having the theme auto switch!

**Comment by numaanashraf** [comment]
Hey everyone! Thanks for your patience. We are actively working on this and should have an ETA on this soon.

**Comment by malekhin** [comment]
+1

**Comment by ankjevel** [comment]
Are you able to give an ETA?

**Comment by schneidewind** [comment]
Hi! is there an update on this?

**Comment by vbasky** [comment]
Is there an update on this ? When can we expect this to be rolled out ?

**Comment by bartekpacia** [comment]
so annyoing

every other app i use has this feature

**Comment by thiagoufg** [comment]
Any news since [Jan 24th, 2022](https://github.com/postmanlabs/postman-app-support/issues/5817#issuecomment-1020359373)? Where is the promised ETA?

**Comment by yaroslav-koval** [comment]
+1

**Comment by NightHunterSV** [comment]
Hi, @numaanashraf, so probably any news on this?

And one more thing: after I've switched from the dark theme to the light, I still have a dark glitch on switching to the Postman from another app:
![image](https://user-images.githubusercontent.com/5138333/169028661-eeb2aeaa-6fe2-440a-ae4e-17bbcb3d9ba7.png)
But in the darkroom, when the dark theme was applied in my OS and manually applied the dark theme to the Postman, I've had the white glitch on switching from another app -- that was very... bright :)
Please, could you also consider fixing this in the scope of this issue?

**Comment by yvoitovych-perfectial** [comment]
+1

**Comment by adriancuadrado** [comment]
> Any news since [Jan 24th, 2022](https://github.com/postmanlabs/postman-app-support/issues/5817#issuecomment-1020359373)? Where is the promised ETA?

@numaanashraf

**Comment by m4k331** [comment]
my eyes are still waiting for this feature 🙏

**Comment by Misha999777** [comment]
+1

**Comment by martin-braun** [comment]
> Hey everyone! Thanks for your patience. We are actively working on this and should have an ETA on this soon.

@numaanashraf Empty words that lack integrity. I don't want to complain, we are thankful for your work and it's a free product after all, but it's always better to promise nothing than to promise the untrue.

**Comment by numaanashraf** [comment]
Hey everyone! This issue is still on our radar and we should be picking this up soon. We don't an ETA just yet - we'll share one as soon as this get prioritised on our near-term roadmap. Thanks for your patience again.

**Comment by fredlemieux** [comment]
<img src="https://media3.giphy.com/media/wi8Ez1mwRcKGI/giphy.gif"/>

**Comment by TomP0** [comment]
+1

**Comment by gabe-cabral** [comment]
Almost 4 years to implement a simple feature that must have been implemented when dark mode was developed... 🤦‍♂️
Can anyone point to the code where we need to look for a PR?

**Comment by adriancuadrado** [comment]
@gcacars I have understood that the Postman app itself is closed-source.

**Comment by martin-braun** [comment]
> Almost 4 years to implement a simple feature that must have been implemented when dark mode was developed... 🤦‍♂️ Can anyone point to the code where we need to look for a PR?

@gcacars 

![image](https://user-images.githubusercontent.com/24259245/184259254-bae605f2-6a04-41fe-8fa8-46cf04229a3a.png)

**Comment by EtzBetz** [comment]
could we at least have a neat quick-toggle in the apps top-bar as a quick-fix for now?

**Comment by yuriisamoienko** [comment]
Please-please-please implement it, my eyes are begging you

**Comment by thiagoufg** [comment]
> Hey everyone! Thanks for your patience. We are actively working on this and should have an ETA on this soon.

ETA, please!

**Comment by aharbavets** [comment]
Absolutely agree, it's very annoying to switch theme manually every day at least 2 times.

Also on Windows and macOS it can be implemented pretty simply:

Windows
-----------
1. Listen for the WMSettingChange event
2. Check `message.section == "ImmersiveColorSet"`
3. If it's true, then set theme according to the value `AppsUseLightTheme` in the Windows registry at the path `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize\`

macOS
--------
1. Subscribe to the `kAppleInterfaceThemeChangedNotification` event using DistributedNotificationCenter
2. Get the current theme (in Swift): 
`let appearanceDescription = NSApplication.shared.effectiveAppearance.debugDescription.lowercased()`
3. if `appearanceDescription.contains("dark")` then set dark theme, else set light theme

Sample app: https://github.com/ruiaureliano/macOS-Appearance

JavaScript (works at least in Chrome, Safari and Firefox)
--------------------------------------------------------
```
window.matchMedia("(prefers-color-scheme: dark)").addListener(
    e => {
        const darkMode = matchMedia('(prefers-color-scheme: dark)').matches;
        // switch theme here
    }
); 
```

**Comment by eriknyk** [comment]
> Hey everyone! Thanks for your patience. We are actively working on this and should have an ETA on this soon.

@thiagoufg Any update about this?

**Comment by Endorphinity** [comment]
Hey, you've already got the code in the previous comment, just get the right engineers and paste it to the proper locations! Come on!

**Comment by Endorphinity** [comment]
Hey, Postman team!

You've already got the code in the previous comment, just get the right engineers and paste it to the proper locations! Come on!

> Absolutely agree, it's very annoying to switch theme manually every day at least 2 times.
> 
> Also on Windows and macOS it can be implemented pretty simply:
> ## Windows
> 
>     1. Listen for the WMSettingChange event
> 
>     2. Check `message.section == "ImmersiveColorSet"`
> 
>     3. If it's true, then set theme according to the value `AppsUseLightTheme` in the Windows registry at the path `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize\`
> 
> 
> ## macOS
> 
>     1. Subscribe to the `kAppleInterfaceThemeChangedNotification` event using DistributedNotificationCenter
> 
>     2. Get the current theme (in Swift):
>        `let appearanceDescription = NSApplication.shared.effectiveAppearance.debugDescription.lowercased()`
> 
>     3. if `appearanceDescription.contains("dark")` then set dark theme, else set light theme
> 
> 
> Sample app: https://github.com/ruiaureliano/macOS-Appearance
> ## JavaScript (works at least in Chrome, Safari and Firefox)
> 
> ```
> window.matchMedia("(prefers-color-scheme: dark)").addListener(
>     e => {
>         const darkMode = matchMedia('(prefers-color-scheme: dark)').matches;
>         // switch theme here
>     }
> ); 
> ```

**Comment by thiagoufg** [comment]
Come on guys, implement this! You promised an ETA. Honor your word!

**Comment by martin-braun** [comment]
I switched to [Hoppscotch](https://hoppscotch.io/) (it's an open-source PWA you can install from your browser), because of this nonsense. I have no shame to suggest you to move away from Postman, because we have reached a point of pure ridiculousness.

A good other reason to switch is to free yourself a little from

![absolutely proprietary](https://user-images.githubusercontent.com/24259245/187676865-a9c519a9-7baa-4286-b680-18274a6f8164.png)

software like Postman.

If this thing was open source, any pioneer would've implemented this years ago. Now all of Postman's user base is suffering from maintainers who either have no time or will to implement the most basic functions that many request and cost almost no time to implement.

It's obvious why people look around corners. Competition is good and Postman is dated for me now. 

I'm out!

**Comment by rafrafek** [comment]
Congrats, it's the only app on my laptop that does not support this. Even notepad on Windows 11 supports auto theme change...

**Comment by rafrafek** [comment]
Seriously it shouldn't be that hard to implement this: https://www.electronjs.org/docs/latest/tutorial/dark-mode

**Comment by aharbavets** [comment]
New major version 10 just came out and there's NO automatic theme switching. We need at least the promised ETA

**Comment by Endorphinity** [comment]
> New major version 10 just came out and there's automatic theme switching. We need at least the promised ETA

… and there's NO automatic theme switching – is that what you meant? At least, I don't see it anywhere in the settings, nor in the release notes.

**Comment by sihaixianyu** [comment]
![image](https://user-images.githubusercontent.com/32222479/194012542-cb064b83-2d38-49c2-96d2-38c14c8a139f.png)

**Comment by johnib** [comment]
Postman team, you remind me my beloved uncle, who had an earring device.
Occasionally, when he didn't want to listen anyone, he would just turn off his device.

Your users are waiting for this feature, make it happen please.

**Comment by aharbavets** [comment]
> > New major version 10 just came out and there's automatic theme switching. We need at least the promised ETA
> 
> … and there's NO automatic theme switching – is that what you meant? At least, I don't see it anywhere in the settings, nor in the release notes.

Yes. Updated the comment

**Comment by antonioacg** [comment]
> I switched to [Hoppscotch](https://hoppscotch.io/) (it's an open-source PWA you can install from your browser), because of this nonsense. I have no shame to suggest you to move away from Postman, because we have reached a point of pure ridiculousness.
> 
> A good other reason to switch is to free yourself a little from
> 
> ![absolutely proprietary](https://user-images.githubusercontent.com/24259245/187676865-a9c519a9-7baa-4286-b680-18274a6f8164.png)
> 
> software like Postman.
> 
> If this thing was open source, any pioneer would've implemented this years ago. Now all of Postman's user base is suffering from maintainers who either have no time or will to implement the most basic functions that many request and cost almost no time to implement.
> 
> It's obvious why people look around corners. Competition is good and Postman is dated for me now.
> 
> I'm out!

Looks awesome!! I will try it. Thanks!!

**Comment by adriancuadrado** [comment]
> I switched to [Hoppscotch](https://hoppscotch.io/) (it's an open-source PWA you can install from your browser), because of this nonsense. I have no shame to suggest you to move away from Postman, because we have reached a point of pure ridiculousness.
> 
> A good other reason to switch is to free yourself a little from
> 
> ![absolutely proprietary](https://user-images.githubusercontent.com/24259245/187676865-a9c519a9-7baa-4286-b680-18274a6f8164.png)
> 
> software like Postman.
> 
> If this thing was open source, any pioneer would've implemented this years ago. Now all of Postman's user base is suffering from maintainers who either have no time or will to implement the most basic functions that many request and cost almost no time to implement.
> 
> It's obvious why people look around corners. Competition is good and Postman is dated for me now.
> 
> I'm out!

Fun fact: Hoppscotch used to be called Postwoman. I think it says a lot about it.

**Comment by gruckion** [comment]
Come on guys... this has been open since 2019 and wouldn't take more than 1 day to implement... Let's be honest it would take a few hours because you already have theme switching and you've got multiple code snippets above. A junior developer can do this and be the hero.

Get this done stop delaying please.

**Comment by malekhin** [comment]
> A junior developer can do this and be the hero.

This is true!

**Comment by diptarag** [comment]
We are working on this feature and targeting a release early next week. I will update here once the feature is published.

**Comment by rafrafek** [comment]
> We are working on this feature and targeting a release early next week. I will update here once the feature is published.

Thanks @diptarag for great news!

**Comment by gruckion** [comment]
> We are working on this feature and targeting a release early next week. I will update here once the feature is published.

Cheers guys.

**Comment by Wyatt-SG** [comment]
ETA?

**Comment by ankjevel** [comment]
@Wyatt-SG [early next week](https://github.com/postmanlabs/postman-app-support/issues/5817#issuecomment-1298536670)

**Comment by joserubicon** [comment]
done !

**Comment by malekhin** [comment]
Oh my god, did this really happen?
We can all get together and celebrate this event.

**Comment by jesseinit** [comment]
FInallyyyyyyy

**Comment by antonioacg** [comment]
Finally!!! Thank you!!! Now only Teams missing. If anyone can vote and help: https://feedbackportal.microsoft.com/feedback/idea/6466f3ed-272e-ec11-b6e6-00224827bbc2

**Comment by eriknyk** [comment]
Thank you! @joserubicon 

_@antonioacg I don't use MS Teams, but just voted to help._

**Comment by diptarag** [comment]
Hey 👋 

This is released as part of Postman v10.1.1
I believe some of you already experienced the new option. Within next few days, everyone should be able to use the feature.

![themesync](https://user-images.githubusercontent.com/2277198/200355093-3bfc258b-c71e-40e4-aa3e-8e0e43226498.gif)
